[TS migration] Migrate 'SettingsAppDownloadLinks' page to TypeScript #25175

hkopser99 opened the pull request:
### Details 

-Migrated JSX component to Typescript 

-Extended MenuItemProps type as DownloadMenuItem type  

-Typed the MenuItems as an array of DownloadMenuItem

-Removed withWindowDimensions/withLocalize PropTypes and compose() from export statement per guidelines on migration  

-Used Hooks for localize translations

-Removed underscore in JSX mapping of menu items. I saw other files converted to TSX that removed it and were getting ESLint warnings. Using the _ is noted as a best practice but given other TSX files had removed or used an ESLint ignore I followed suit. I did not get ESLint warnings, so I did not need the ignore comment. 

-I removed onKeyDown attribute and blur() from MenuItem. I did not see this elsewhere in the codebase and the removal had no impact on the page’s functionality or appearance. It appeared antiquated but please advise if there is a need for it that I am missing)  

-This component receives props from Navigation but I do not see elsewhere in the codebase that we are explicitly listing navigation related props when migrating to TS (nav/routing props should inherently apply to all pages presented via Navigation) 

-A TS warning made me list translationKey in DownloadMenuItem as type TranslationPaths. The warning only appeared when not providing translate from useLocalize() with a string literal. Others working on TS migrations may want to be aware of this as a solution. 

-No other components or other files in the codebase import this file so changing from .js to .tsx should not require revisions to import statements. This component is routed to by using its displayName and that was not changed. 

-One item to note is that on the iOS simulator there is no App Store app, so it does not know how to handle the download iOS app link click (see screenshots on iOS Native and iOS mWeb for reference). The link supplied is correct, and it works everywhere else.

<!-- TS Migration AppDownloadLinks --> 

### Fixed Issues 
$ #25175 

PROPOSAL:
$ https://github.com/Expensify/App/issues/25175#issuecomment-1884662977  

Proposed to migrate file to Typescript and convert file type to TSX following Expensify migration best practices as well as testing on all platforms to confirm page functionality remains the same. 

 
### Tests 

I ran the app on each platform, confirmed each App Download Link leads to the intended URL. I also confirmed that navigation back to the About page operates as intended when leaving the AppDownloadLinks page. 
  

- [X] Verify that no errors appear in the JS console 

  

### Offline tests 

The links work offline but will not present web pages with content or the MacOS App Download option until internet connection has been reestablished. During this time standard Safari/Chrome warnings are presented. 
  

### QA Steps 

1. Login and then visit Settings->About->App Download Links
2. Click/tap each link to confirm they are sending the user to the right download website (e.g. Google Play, App Store/Apple Store website, or desktop download for MacOS App)
3. Navigate back to the About page when done to confirm navigation functions as intended
  

- [X] Verify that no errors appear in the JS console 

  

### PR Author Checklist 

<!-- 

This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged! 

--> 

  

- [X] I linked the correct issue in the `### Fixed Issues` section above 

- [X] I wrote clear testing steps that cover the changes made in this PR 

    - [X] I added steps for local testing in the `Tests` section 

    - [X] I added steps for the expected offline behavior in the `Offline steps` section 

    - [X] I added steps for Staging and/or Production testing in the `QA steps` section 

    - [X] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct) 

    - [X] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline) 

    - [X] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability). 

- [X] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms) 

- [X] I ran the tests on **all platforms** & verified they passed on: 

    - [X] Android: Native 

    - [X] Android: mWeb Chrome 

    - [X] iOS: Native 

    - [X] iOS: mWeb Safari 

    - [X] MacOS: Chrome / Safari 

    - [X] MacOS: Desktop 

- [X] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed) 

- [X] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code)) 

    - [X] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`) 

    - [X] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`. 

    - [X] I verified that comments were added to code that is not self explanatory 

    - [X] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing. 

    - [X] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60) 

      - [X] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message: 

    - [X] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68) 

    - [X] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy. 

    - [X] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README. 

    - [X] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed 

- [X] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers 

- [X] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md) 

- [X] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected) 

- [X] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests) 

- [X] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such 

- [X] I verified that if a function's arguments changed that all usages have also been updated correctly 

- [X] If any new file was added I verified that: 

    - [X] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory 

- [X] If a new CSS style is added I verified that: 

    - [X] A similar style doesn't already exist 

    - [X] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`) 

- [X] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic. 

- [X] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases) 

- [X] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected. 

- [X] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account. 

- [X] If the PR modifies the form input styles: 

    - [X] I verified that all the inputs inside a form are aligned with each other. 

    - [X] I added `Design` label so the design team can review the changes. 

- [X] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page. 

- [X] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps. 

  

### Screenshots/Videos 

<details> 

<summary>Android: Native</summary> 

  

<img width="353" alt="ExpensifyADLTSMigAndroidEmulatorPage" src="https://github.com/Expensify/App/assets/144170694/a73c3131-2bb0-46f7-b8ba-c4ea32a23519"> 

<img width="347" alt="ExpensifyADLTSMigAndroidEmulatorAndroidAppLink" src="https://github.com/Expensify/App/assets/144170694/a7aba7c3-bf09-42bb-bbbb-8e0a2ec110af"> 

<img width="348" alt="ExpensifyADLTSMigAndroidEmulatoriOSAppLink" src="https://github.com/Expensify/App/assets/144170694/bfe9e815-7973-4a77-afd3-e13573b0df19"> 

<img width="353" alt="ExpensifyADLTSMigAndroidEmulatorMacOSAppLink" src="https://github.com/Expensify/App/assets/144170694/6bf49d26-4496-4417-9d1f-a4a4eeb4bd3e"> 

</details> 

  

<details> 

<summary>Android: mWeb Chrome</summary> 

  

<img width="352" alt="ExpensifyADLTSMigChromeWebPage" src="https://github.com/Expensify/App/assets/144170694/5c82500c-73dc-4e0b-80a3-fd20b26e72b1"> 

<img width="354" alt="ExpensifyADLTSMigChromeWebGooglePlayLink" src="https://github.com/Expensify/App/assets/144170694/06666409-8637-4523-82fd-39677d500b17"> 

<img width="348" alt="ExpensifyADLTSMigChromeWebiOSApp" src="https://github.com/Expensify/App/assets/144170694/db94703b-cd71-4015-909a-5ae3cf27e0a4"> 

<img width="345" alt="ExpensifyADLTSMigChromeWebMacOSLink" src="https://github.com/Expensify/App/assets/144170694/a6489b9d-e7fb-46ea-a2ae-23967325cc92"> 

  

</details> 

  

<details> 

<summary>iOS: Native</summary> 

  

![ExpensifyADLTSMigiOSNative](https://github.com/Expensify/App/assets/144170694/90924bc2-7805-43d1-b9f7-b6764c909ca4) 

![ExpensifyADLTSMigiOSNativeAndroidLink](https://github.com/Expensify/App/assets/144170694/77672d4b-83ef-4258-aacf-469cc72f35e0) 

![ExpensifyADLTSMigiOSNativeiOSAppLink](https://github.com/Expensify/App/assets/144170694/c50d9987-67a7-4e86-b2f1-d96d20172ffe) 

![ExpensifyADLTSMigiOSNativeMacOSLink](https://github.com/Expensify/App/assets/144170694/2c7a96c0-8a29-40c3-8f77-995467d63eae) 

  

</details> 

  

<details> 

<summary>iOS: mWeb Safari</summary> 

  

<img width="356" alt="ExpensifyADLTSMigmSafariPage" src="https://github.com/Expensify/App/assets/144170694/d14acf39-a023-4214-bf7e-24fdd7d8db97"> 

<img width="356" alt="ExpensifyADLTSMigMSafariAndroidLink" src="https://github.com/Expensify/App/assets/144170694/64b53810-9b2b-41bf-a5a9-409500939bc3"> 

<img width="366" alt="ExpensifyADLTSMigMSafariiOSAppLink" src="https://github.com/Expensify/App/assets/144170694/ca36a656-42e8-40a0-b9d3-0fdcaafec624"> 

<img width="350" alt="ExpensifyADLTSMigMSafariMacOSAppLink" src="https://github.com/Expensify/App/assets/144170694/dfd7cbb9-ef63-46dd-8db2-7a1dbf9bd48c"> 

  

</details> 

  

<details> 

<summary>MacOS: Chrome / Safari</summary> 

  

<img width="1422" alt="ExpensifyADLTSMigDesktopPg" src="https://github.com/Expensify/App/assets/144170694/47170449-a138-4abc-a467-7b8a1902e2d0"> 

<img width="1507" alt="ExpensifyADLTSMigDesktopLinksWorking" src="https://github.com/Expensify/App/assets/144170694/883fff73-3ca8-4934-a56d-819050cc7cab"> 

  

</details> 

  

<details> 

<summary>MacOS: Desktop</summary> 

  

<img width="1512" alt="ExpensifyADLTSMigMacOSAppPage" src="https://github.com/Expensify/App/assets/144170694/3ba0f0a4-f1c8-4e0f-b3a8-04696f0d5f1e"> 

<img width="1512" alt="ExpensifyADLTSMigMacOSAppLinksWorking" src="https://github.com/Expensify/App/assets/144170694/7bbb7391-d240-4c3a-a19c-e6e551d67a11"> 

  

  

</details>

hkopser99 committed:
TS Migration #25175: Update and rename AppDownloadLinks.js to AppDownloadLinks.tsx

Updates related to Typescript Migration #25175. 

-Migrated component to Typescript
-Extended MenuItemProps type as DownloadMenuItem type
-Typed menu items as an array of DownloadMenuItems
-Removed withWindowDimensions/withLocalize PropTypes and compose() from export per guidelines on migration
-Used Hooks for localize instead for translation
-Removed _ in JSX mapping of menu items. Saw other TSX files that removes and I was getting ESLint warnings. _ is noted as a best practice but given other TSX files have removed I followed suit. I also saw old ESlint ignores being used when removing _ but I didn't get flagged without this.
-Removed onKeyDown attribute and blur() from MenuItem. I saw this nowhere else in the codebase and the removal had no impact on app use functionality (please advise if there's something I am missing but it appeared antiquated)
-This component receives props from Navigation but I do not see elsewhere in the codebase that we are explicitly listing these as props via typescript (navigation props should inherintly apply to all pages)
-TS warnings made me list translationKey as type TranslationPaths (warning only appears when not providing the translate from useLocalize() with a '' - others working on TS migrations may want to know this)
-No ESlint warnings
-No VSCode warnings or errors
-No errors when testing
-Only issue was on iOS simulator there's no App Store app so it doesn't know how to handle the download iOS click but the link is fine and it works everywhere else.
-No other components or other files in codebase import this file so change from .js to .tsx should not require changes to imports anywhere. This component is routed to via its name as with other pages in codebase.

hkopser99 commented:
I have read the CLA document and I hereby sign the CLA

hkopser99 commented:
recheck

Santhosh-Sellavel commented:
Can you fix the lint issues?

Santhosh-Sellavel commented:
Please link the issue properly in the description. Also, can you check and update descriptions correctly there few typos

hkopser99 committed:
Update AppDownloadLinks.tsx

hkopser99 commented:
@Santhosh-Sellavel ran Prettier so the lint issues should be resolved. Do I need to tell the bot to recheck as I did after signing the agreement? I also fixed the description and the linking to the fix along with a few other revisions for clarity and conciseness. I found another recent PR and tried to make mine more like that so you have an easier review. If you notice anything else missing just let me know. Its my first PR for Expensify so I want to make sure you have everything you need.

Santhosh-Sellavel commented:
@Julesssss 
Can you initiate workflow checks?

hkopser99 commented:
@Santhosh-Sellavel I saw an error occurred so I ran the perf-tests locally and didn't see it on my local environment. I found a few different threads in slack from Nov 2023 - [Feb](https://expensify.slack.com/archives/C01GTK53T8Q/p1706732738472299) of this year mentioning perf-test failures where changes were unrelated to the failure. At least one was resolved just by re-running the tests. Is this an issue you've encountered before?

kubabutkiewicz commented:
@hkopser99 for fixing Reassure Perfomance tests just marge main to your branch

hkopser99 commented on src/pages/settings/AppDownloadLinks.tsx:
```diff
@@ -1,75 +1,75 @@
+    const {translate} = useLocalize();
+    const popoverAnchor = useRef<View>(null);
+
+    type DownloadMenuItem = MenuItemProps & {
```
Thanks, I'll move that here and will be sure to move type declarations outside of component scope going forward.

hkopser99 commented on src/pages/settings/AppDownloadLinks.tsx:
```diff
@@ -1,75 +1,75 @@
-                title={props.translate('initialSettingsPage.aboutPage.appDownloadLinks')}
-                onBackButtonPress={() => Navigation.goBack()}
+                title={translate('initialSettingsPage.aboutPage.appDownloadLinks')}
+                onBackButtonPress={() => Navigation.goBack(ROUTES.SETTINGS_ABOUT)}
```
My logic when I was working on the change was to make the navigation destination explicit. Now that I look at it there's no benefit to this. Navigation.goBack() works as intended here and removing the explicit destination makes the code more reusable if we ever want to present the AppDownloadLinks page from another screen in the future. If approved I will amend to remove ROUTES.SETTINGS_ABOUT as an argument and will then also be able to remove the ROUTES import statement above as this was the only use of ROUTES.

situchan replied:
Keep original logic if possible.

hkopser99 committed:
Update AppDownloadLinks.tsx

Moved Type declaration out of component scope.

hkopser99 committed:
Merge branch 'Expensify:main' into patch-1

Santhosh-Sellavel commented:
@Julesssss Can you rerun workflow tests

Julesssss commented:
> @Julesssss Can you rerun workflow tests

Yep, done

situchan commented on src/pages/settings/AppDownloadLinks.tsx:
```diff
@@ -1,75 +1,75 @@
-            iconRight: Expensicons.NewWindow,
-            action: () => {
+            openAppDownloadLink: () => {
                 Link.openExternalLink(CONST.APP_DOWNLOAD_LINKS.ANDROID);
             },
-            link: CONST.APP_DOWNLOAD_LINKS.ANDROID,
+            downloadLink: CONST.APP_DOWNLOAD_LINKS.ANDROID,
```
Any reason for renaming? action > openAppDownloadLink, link > downloadLink

hkopser99 replied:
Yes, I was following my understanding of the Style document in the Naming Conventions->Event Handlers section found [here](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#naming-conventions). I then adjusted link to downloadLink to be more descriptive. Please let me know if 'action' and 'link' are preferred to the more descriptive names.

situchan replied:
original names are fine to me

situchan reviewed (changes requested):
Please merge main as 1k+ commits are behind

situchan commented:
@hkopser99 I am not sure what happened with your author checklist. Please check any other PRs

<img width="676" alt="Screenshot 2024-02-14 at 3 32 26 AM" src="https://github.com/Expensify/App/assets/108292595/73f46872-a096-4db1-b938-2504e1f0aec8">


---

Add Tests / QA Steps

hkopser99 committed:
Update AppDownloadLinks.tsx

Amended openAppDownloadLink and downloadLink in DownloadMenuItem back to action and link.

Returned to original, simplified logic for Navigation.goBack()

hkopser99 committed:
Merge branch 'Expensify:main' into patch-1

situchan commented:
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

Julesssss commented:
Hey @situchan, no rush but I think we're just waiting on the final device testing and screen captures, right?

situchan commented:
> Hey @situchan, no rush but I think we're just waiting on the final device testing and screen captures, right?

right

situchan reviewed (approved):
:shipit:

Julesssss commented:
Hi @hkopser99 would you mind merging `main` again to confirm there aren't any conflicts please

hkopser99 committed:
Merge branch 'Expensify:main' into patch-1

hkopser99 commented:
@Julesssss just merged!

Julesssss commented:
Thanks

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

hkopser99 commented:
@Julesssss thank you and thanks for letting me work on this one. Had some bumps along the way getting up and running w/ the codebase/dependencies and my laptop but thankfully the actual migration piece wasn't tricky :) Do you have any more of these or other stuff to work on? Would love to do more

Julesssss commented:
No worries! You can find all our open issues [here](https://github.com/Expensify/App/issues?q=is%3Aissue+label%3AExternal+is%3Aopen), but some of them will have already been assigned to contributors.

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/7966219100) to staging by https://github.com/Julesssss in version: 1.4.43-0 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/8055338936) to production by https://github.com/puneetlath in version: 1.4.43-20 🚀

platform | result
---|---
🤖 android 🤖|success ✅
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅
🕸 web 🕸|success ✅
